flaml/searcher/blendsearch.py

```diff
@@ -39,9 +39,11 @@ def __init__(self,
                  min_resource: Optional[float] = None,
                  max_resource: Optional[float] = None,
                  reduction_factor: Optional[float] = None,
-                 resources_per_trial: Optional[dict] = None,
                  global_search_alg: Optional[Searcher] = None,
-                 mem_size: Callable[[dict], float] = None,
+                 config_constraints: Optional[
+                     List[Tuple[Callable[[dict], float], str, float]]] = None,

```

Comment:
list of arguments

Reply:
This is hard to parse: List[Tuple[Callable[[dict], float], str, float]]]

What about introducing a lightweight dataclass to manage these? e.g.

@dataclass
class Constraint:
  metric_function: Callable[Dict, float]
  inequality: str
  value: float

and replace type with Optional[List[Constraint]]. This would also make the code more readable below.